data/src/main/java/com/example/data/di/DatabaseModule.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.data.di
+
+import android.app.Application
+import androidx.room.Room
+import com.example.data.database.AlpakiDB
+import com.example.data.database.dao.DreamCategoryDao
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ApplicationComponent
+import javax.inject.Singleton
+
+@InstallIn(ApplicationComponent::class)
+@Module
+object DataBaseModule {
+    @Provides
+    @Singleton
+    fun provideAlpakiDb(application: Application): AlpakiDB {
+        return Room.databaseBuilder(application, AlpakiDB::class.java, "AlpakiDatabase")
+            .fallbackToDestructiveMigration()
+            .allowMainThreadQueries()
```

Comment:
Czy to jest nam potrzebne? O ile dobrze pamiętam, Dao pod maską używają backgroundowych wątków dla suspendowalnych metod. Jeżeli tak nie jest, to i tak sami powinniśmy obsłużyć zmianę wątku a nie wywoływać te metody z głównego.

Reply:
masz racje, musiało mi sie przez przypadek zaciągnąć ;)